packages/picasso/src/Badge/Badge.tsx

```diff
@@ -1,36 +1,52 @@
-import React, { forwardRef } from 'react'
+import React, { Children, forwardRef, ReactNode } from 'react'
 import { makeStyles, Theme } from '@material-ui/core/styles'
 import cx from 'classnames'
-import { BaseProps, TextLabelProps } from '@toptal/picasso-shared'
+import { Badge as MuiBadge } from '@material-ui/core'
+import { BaseProps, TextLabelProps, useTitleCase } from '@toptal/picasso-shared'
 
 import styles from './styles'
+import { toTitleCase } from '../utils'
 
 type VariantType = 'white' | 'red'
 type SizeType = 'medium' | 'small' | 'large'
 
 export interface Props extends BaseProps, TextLabelProps {
-  /** The `Badge` content */
-  content: number
-  /** Variant of the `Badge` */
+  /**
+   * The `Badge` content
+   */
+  content: number | string
+
+  /**
+   * Variant of the `Badge`
+   */
   variant?: VariantType
-  /** Size of the `Badge` */
+
+  /**
+   * Size of the `Badge`
+   */
   size?: SizeType
+
   /**
    * Max count to show. By default 9 for small size, 99 for other sizes
    */
   max?: number
+
+  /**
+   * The badged will be overlaid on it's children 
+   */
+  children?: ReactNode
 }
 
 const useStyles = makeStyles<Theme>(styles, { name: 'PicassoBadge' })
 
-const thresholds = {
+const thresholds: { [k in SizeType]: number } = {
```

Comment:
I just wanted to know if you have better reasoning or is it just a preference for not using Record here?
```suggestion
const thresholds: Record<SizeType, number> = {
```

Reply:
No reasoning, both should represent the same type. `Record` looks terser in fact, I will refactor to that